### DISCUSSION TITLE
Is there an execution alias for Terminal Preview?

### DISCUSSION THREAD

**Discussion by devlie** [microsoft/terminal#14856] [Q&A] [answered]
wt.exe seems to be coupled with the release version only. I'm trying to avoid hardcoding the preview exe path in my script :)

**Comment by DHowett** [comment, marked as answer]
There is no distinct execution alias for Terminal Preview, as we wanted to make sure that users could opt in for all invocations of `wt` pointing at one or the other. If your script requires Preview, (1) that's cool! and (2) I'd love to hear about why 😁 

In general, whether Preview or Stable is selected was always intended to be an end-user choice.

**Reply by devlie** [reply]
I have some PowerShell scripts that need to call `Start-Process` to launch some long-running background processes. Today that creates new PS window outside of Terminal, occupying my precious taskbar space :), so I thought I'm going to have it launch into a tab instead. I found that I have to hardcode the preview exe path because `wt` only points to Release version. I'm using Preview daily but prefer keeping Release installed in case of blocking bugs.

**Reply by DHowett** [reply]
Ah! In that case, you might be better served by changing one of the following sets of settings:

- Reconfigure which version of Terminal `wt.exe` points to
  - Start Menu search for "app execution aliases", and choose "manage app execution aliases"
  - Turn on the one for Terminal Preview, which will turn off the other one for Terminal automatically:
  - <img width="344" alt="image" src="https://user-images.githubusercontent.com/189190/219131762-786e5ce3-e60c-45d6-98c2-c1b632261895.png">

.. or ..

- Change your default console host to Terminal Preview and change the new instance behavior to "most recent window on current desktop"
  - <img width="524" alt="image" src="https://user-images.githubusercontent.com/189190/219131977-a7f559d7-0570-4c54-a0bc-20e99618ae39.png">
  - <img width="606" alt="image" src="https://user-images.githubusercontent.com/189190/219132022-c5b5f760-c092-477a-ba82-4f4be7d70830.png">

**Reply by devlie** [reply]
Thanks, I gave them a try, and they work for `start-process cmd.exe` but not `start-process powershell`. I tried changing the default profile to PS5 too, but no dice.

**Reply by zadjii-msft** [reply]
> but not `start-process powershell`

Hmmm..... that's peculiar. Does doing that open an _elevated_ powershell? Maybe that exe was set to always run as admin?

**Reply by devlie** [reply]
> Hmmm..... that's peculiar. Does doing that open an _elevated_ powershell? Maybe that exe was set to always run as admin?

Sorry, I didn't realize that the privilege matters. I was running Terminal elevated because my ps1 script requires elevation. 

Here's the updated outcome:

Privilege | Outcome
-- | --
Terminal not elevated | Both `start-process powershell` and `start-process cmd` open in new tab.
Terminal elevated | Both open in new window.

Is there a way to make it work when elevated?

**Reply by zadjii-msft** [reply]
> Is there a way to make it work when elevated?

There sure isn't - we're working with the platform team to figure that out. It's a Packaged COM issue that we're "tracking" externally in #10276
